server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -77,13 +77,19 @@ export class AssetController {
     @UploadedFiles() files: { assetData: Express.Multer.File[]; livePhotoData?: Express.Multer.File[] },
     @Body(ValidationPipe) createAssetDto: CreateAssetDto,
     @Response({ passthrough: true }) res: Res,
+    @Query('key') sharedKey?: string,
```

Comment:
We shouldn't be manually injecting stuff here for authentication purposes, it should go through the `AuthUserDto`.